docs/en/amls.md

```diff
@@ -0,0 +1,665 @@
+		lon_gps_loop_add = (float)(lon_gps_actual - lon_gps_previous) / 10.0;
+
+		// Set the l_lat_gps variable to the previous latitude value
+		l_lat_gps = lat_gps_previous;
+		// Set the l_lon_gps variable to the previous longitude value
+		l_lon_gps = lon_gps_previous;
+
+		// Remember the new latitude value in the lat_gps_previous variable for the next loop
+		lat_gps_previous = lat_gps_actual;
+		// Remember the new longitude value in the lat_gps_previous variable for the next loop
+		lon_gps_previous = lon_gps_actual;
+	}
+
+	// If the line starts with SA and if there is a GPS fix we can scan the line for the fix type (none, 2D or 3D)
+	if (incomming_message[4] == 'S' && incomming_message[5] == 'A')
+		fix_type = (int)incomming_message[9] - 48;
+
+}
+```
+
+### 1.3. Set current waypoint
+
+When required data is received the main magic happens. To enable maintaining of the current position it will be enough to set the flag `waypoint_set = 1;` and set current coordinates as a waypoint:
+
+```cpp
+l_lat_waypoint = l_lat_gps;
+l_lon_waypoint = l_lon_gps;
+```
+
+After that, the calculation of the error in the coordinates will begin, correction works with the help of a PD - regulator. For D - component we use rotation memory.
+
+### 1.4. Waypoint edit (To fly to waypoints)
+
+If you just set the new `l_lat_waypoint` and `l_lon_wayoint`, which are located at a great distance from the drone, the drone will not be able to fly normally and stabilize at these coordinates. For smooth adjustments `l_lat_gps_float_adjust` and `l_lon_gps_float_adjust` can be used. These are `float` variables, changing which will smoothly shift `l_lat_waypoint` and `l_lon_waypoint`.
+
+For example, if in the main loop you will constantly add a certain value to these variables:
+
+```cpp
+l_lat_gps_float_adjust += 0.0015;
+```
+
+With set waypoint, the drone will move smoothly in the given direction.
+In the future, this will be used for the smooth drone's acceleration and deceleration while moving to its destination.
+
+### 1.5. Waypoint stabilization
+
+```cpp
+if (waypoint_set == 1) {                           
+	//If the waypoints are stored
+	
+	// Adjust l_lat_waypoint
+	if (l_lat_gps_float_adjust > 1) {
+		l_lat_waypoint++;
+		l_lat_gps_float_adjust--;
+	}
+	if (l_lat_gps_float_adjust < -1) {
+		l_lat_waypoint--;
+		l_lat_gps_float_adjust++;
+	}
+
+	// Adjust l_lon_waypoint
+	if (l_lon_gps_float_adjust > 1) {
+		l_lon_waypoint++;
+		l_lon_gps_float_adjust--;
+	}
+	if (l_lon_gps_float_adjust < -1) {
+		l_lon_waypoint--;
+		l_lon_gps_float_adjust++;
+	}
+
+	// Calculate the latitude error between waypoint and actual position
+	gps_lon_error = l_lon_waypoint - l_lon_gps;
+	// Calculate the longitude error between waypoint and actual position
+	gps_lat_error = l_lat_gps - l_lat_waypoint;
+
+	// Subtract the current memory position to make room for the new value
+	gps_lat_total_avarage -= gps_lat_rotating_mem[gps_rotating_mem_location];
```

Comment:
average? But I guess it's to difficult to change the code.

Reply:
This is not quite an average. This rotating memory is used not only here. This will be replaced by classic D and I-factors in the future.

Reply:
No, I mean the words spells average, not avarage.